### ISSUE TITLE
Conv2d about Weights_training 

### ISSUE THREAD

**Issue by ChenYang1024** [KaidiXu/auto_LiRPA#10]
We try to implement Lenet5 with weights_training.py. First, we add a network Lenet5 with perturb in feedword.py, the network contains three conv2d and two linear, and when we use this network, the loss is NAN. Then, we find the bug is from weights_training.py when we use CROWN-IBP to calculate ub, the tensor value of ub is negative, and the torch.log(ub) is NAN. Then we remove the perturb of Conv2d,  and we use the normal Conv2d layer and the linear layer with perturb, we find the tensor value of ub is also negative. Finally, we remove the Conv2d layer and only remain the perturb linear layer, the loss is normal.
Could you please tell me about how to deal with the negative ub value? How about we use abs() to change the ub to positive? 
Thank you!

**Comment by huanzhang12** [comment]
@1406753105 We have just made a new release with better pooling layer support. You can try the latest version of auto_LiRPA and see if the problem is still there. If the new release does not fix your issue, please kindly share a sample code that can reproduce this problem with us. Thank you!
